tests/Resources/Page/ODH/Prometheus/Prometheus.robot

```diff
@@ -17,6 +17,29 @@ Run Query
     # Log To Console    ${resp.json()}
     [Return]    ${resp}
 
+Run Range Query
+    [Documentation]    Runs a prometheus range query, in order to obtain the result of a PromQL expression over a given time range. More info at:
+    ...                - https://promlabs.com/blog/2020/06/18/the-anatomy-of-a-promql-query
+    ...                - https://prometheus.io/docs/prometheus/latest/querying/api/#range-queries
+    [Arguments]    ${pm_query}    ${pm_url}    ${pm_token}    ${interval}=12h     ${steps}=172
+    ${time} =    Get Start Time And End Time  interval=${interval}
+    ${pm_headers}=    Create Dictionary    Authorization=Bearer ${RHODS_PROMETHEUS_TOKEN}
+    ${resp}=    RequestsLibrary.GET    url=${RHODS_PROMETHEUS_URL}/api/v1/query_range?query=${pm_query}&start=${time[0]}&end=${time[1]}&step=${steps}
+    ...    headers=${pm_headers}    verify=${False}
+    Status Should Be    200    ${resp}
+    # Log To Console    ${resp.json()}
+    [Return]    ${resp}
+
+Get Start Time And End Time
+    [Documentation]     Returns start and end time for Query range from current time
+    [Arguments]         ${interval}   # like 12h  7 days etc
+    ${end_time} =  Get Current Date
+    ${end_time} =  BuiltIn.Evaluate  datetime.datetime.fromisoformat("${end_time}").timestamp()
```

Comment:
Did you find the problem when running this line in Jenkins?

Reply:
This time no
I have not faced any error